cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -0,0 +1,462 @@
+        self.coord = Coord(x_value=0, y_value=0)
+        self.graph_width = 6
+        self.graph_height = 6
+        self.logs_width = 24
+        self.logs_height = 6
+        self.empty_section = True
+        self.dashboard = None
+
+        self._add_resources()
+
+    # -- Utility methods --------------------------------------------------------------------------------------------- #
+
+    @property
+    def _stack_region(self):
+        return core.Stack.of(self).region
+
+    # -- Resources --------------------------------------------------------------------------------------------------- #
+
+    def _add_resources(self):
+
+        # Initialize a cloudwatch dashboard
+        self.cloudwatch_dashboard = cloudwatch.Dashboard(
+            scope=self.stack_scope, id="CloudwatchDashboard", dashboard_name=self.dashboard_name
+        )
+
+        # Create a text widget for title "Head Node EC2 metrics"
+        self._add_text_widget("# Head Node EC2 Metrics")
+
+        # Add head node instance graphs
+        self._add_head_node_instance_metrics_graphs()
+
+        ebs_metrics = [
+            GeneralMetric("Read/Write Ops", ["VolumeReadOps", "VolumeWriteOps"]),
+            GeneralMetric("Read/Write Bytes", ["VolumeReadBytes", "VolumeWriteBytes"]),
+            GeneralMetric("Total Read/Write Time", ["VolumeTotalReadTime", "VolumeTotalWriteTime"]),
+            GeneralMetric("Queue Length", ["VolumeQueueLength"]),
+            GeneralMetric("Idle Time", ["VolumeIdleTime"]),
+        ]
+
+        conditional_ebs_metrics = [
+            GeneralMetric("Consumed Read/Write Ops", "VolumeConsumedReadWriteOps", ["io1", "io2", "gp3"]),
+            GeneralMetric("Throughput Percentage", "VolumeThroughputPercentage", ["io1", "io2", "gp3"]),
+            GeneralMetric("Burst Balance", "BurstBalance", ["gp2", "st1", "sc1"]),
+        ]
+
+        # Add EBS and RAID metrics graphs
+        for storage_type, title in [
+            (SharedStorageType.EBS, "## EBS Metrics"),
+            (SharedStorageType.RAID, "## RAID Metrics"),
+        ]:
+            if len(self.storage_resources[storage_type]) > 0:
+                self._add_volume_metrics_graphs(title, storage_type, ebs_metrics, conditional_ebs_metrics)
+
+        # Add EFS metrics graphs
+        if len(self.storage_resources[SharedStorageType.EFS]) > 0:
+            self._add_efs_metrics_graphs()
+
+        # Add FSx metrics graphs
+        if len(self.storage_resources[SharedStorageType.FSX]) > 0:
+            self._add_fsx_metrics_graphs()
+
+        # Head Node logs, if CW Logs are enabled
+        if self.config.is_cw_logging_enabled:
+            self._add_cw_log()
+
+    def _update_coord(self, d_x, d_y):
+        """Calculate coordinates for the new graph."""
+        self.coord.x_value = self.coord.x_value + d_x
+        if self.coord.x_value > 24:  # the longest width allowed is 24
+            self.coord = Coord(0, self.coord.y_value + d_y)  # updates both x and y values
```

Comment:
you don't need to instantiate another coord.
You can do something like:
```python
x_value = self.coord.x_value + d_x
self.coord.x_value = 0 if x_value > 24 else x_value
self.coord.y_value += d_y
```

Reply:
Since it will change both x and y value of x_value>24,
 now I change it into:
if x_value > MAX_WIDTH:  # updates both x and y values       
            self.coord.x_value = 0 
            self.coord.y_value += d_y